includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -631,7 +634,7 @@ public function get_payment_fields_js_config() {
 			'locale'                         => WC_Payments_Utils::convert_to_stripe_locale( get_locale() ),
 			'isUPEEnabled'                   => WC_Payments_Features::is_upe_enabled(),
 			'isSavedCardsEnabled'            => $this->is_saved_cards_enabled(),
-			'isPlatformCheckoutEnabled'      => WC_Payments_Features::is_platform_checkout_eligible() && 'yes' === $this->get_option( 'platform_checkout', 'no' ),
+			'isPlatformCheckoutEnabled'      => $platform_checkout_util->should_enable_platform_checkout( $this ),
```

Comment:
This flag handles the checkout email input https://github.com/Automattic/woocommerce-payments/blob/0ab629c971bfe7ab61646ca96cbde786ff121b6f/client/checkout/classic/index.js#L561-L563.

It will now also check if subscription product is in cart before enabling.